component/mimir/rules/events.go

```diff
@@ -0,0 +1,261 @@
+package rules
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/ghodss/yaml"
```

Comment:
`github.com/ghodss/yaml` is necessary here because it's used to marshal from structs that only have json annotations. Rather than coming up with a solution I just copied what the prometheus-operator does to maintain 100% compatibility: https://github.com/prometheus-operator/prometheus-operator/blob/3ee8ce54baea4e99c1e2dfe6bd6d0b6b8b265e76/pkg/operator/rules.go#L21

Reply:
I'll add a small comment to that line to clarify

Reply:
👍 Thanks!